cpp/dolfinx/refinement/utils.cpp

```diff
@@ -86,16 +86,16 @@ xt::xtensor<double, 2> create_new_geometry(
 
   // Copy over existing mesh vertices
   xtl::span<const double> x_g = mesh.geometry().x();
-
+  const std::size_t gdim = mesh.geometry().dim();
   const std::size_t num_vertices = map_v->size_local();
   const std::size_t num_new_vertices = local_edge_to_new_vertex.size();
   xt::xtensor<double, 2> new_vertex_coordinates(
-      {num_vertices + num_new_vertices, 3});
+      {num_vertices + num_new_vertices, gdim});
 
   for (std::size_t v = 0; v < num_vertices; ++v)
   {
     const int pos = 3 * vertex_to_x[v];
-    for (std::size_t j = 0; j < 3; ++j)
+    for (std::size_t j = 0; j < gdim; ++j)
```

Comment:
It's better for performance to have a fixed loop bound here.

Reply:
Even if it means that we need to do a strided copy when returning the data?

Reply:
Make everything 3D.

Reply:
We need to remove the last dimensions before passing this to the geometry constructor, as it computes the gdim from the shape of the input.